### PR TITLE
DM-43124: Add redis store of unfurl events to debounce replies

### PR DESCRIPTION
**Pull request by jonathansick**
- The SlackUnfurlEventStore is a hash Redis store where keys (formatted via SlackUnfurlEventKey) identify events where a token (e.g. Jira issue key) has been "unfurled" in a given channel and thread timestamp.
- Adding items into the SlackUnfurlEventStore is embedded into the DomainUnfurler. Events are added as part of the send_reply method, and the is_recently_unfurled method lets clients know if a ticket has been recently unfurled.

### CODE REVIEW

File: kafka-compose.yaml [hunk 1]
```diff
@@ -53,3 +53,12 @@ services:
       KAFKA_BROKERCONNECT: "kafka1:19092"
     depends_on:
       - "kafka1"
+
+  redis:
```
Comment (athornton) [inline kafka-compose.yaml:57]: Why not use the (opinionated, but its opinions are, I think correct for us) redis chart in lsst-sqre/charts ?

**Review by athornton — APPROVED** [review]
Looks reasonable, but I am confused why you didn't use the lsst-sqre/charts redis chart.  If we're not supposed to use that one, then I need to change checkerboard, but I thought the purpose of that chart was to give us a shared opinionated redis whose opinions matched the usual SQuaRE use case.

File: kafka-compose.yaml (hunk 1 above)
Comment (athornton) [reply kafka-compose.yaml:57]: Ah, OK, because this is for the test suite, not the _actual_ orchestration.

Comment (jonathansick) [reply kafka-compose.yaml:57]: Right, Phalanx deployment/charts are at https://github.com/lsst-sqre/phalanx/pull/3014 and we're using the internal redis chart there.

**Pull request merged**